webapp/channels/src/components/channel_header/channel_header_title_favorite.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useRef, useCallback} from 'react';
+import {useSelector, useDispatch} from 'react-redux';
+import {FormattedMessage, useIntl} from 'react-intl';
+import OverlayTrigger from 'components/overlay_trigger';
+import Tooltip from 'components/tooltip';
+import {Constants } from 'utils/constants';
+import {favoriteChannel, unfavoriteChannel} from 'mattermost-redux/actions/channels';
+import {getCurrentChannel, isCurrentChannelFavorite} from 'mattermost-redux/selectors/entities/channels';
+
+type Props = {}
+
+const ChannelHeaderTitleFavorite = ({}: Props) => {
+    const intl = useIntl();
+    const dispatch = useDispatch()
+    const isFavorite = useSelector(isCurrentChannelFavorite);
+    const channel = useSelector(getCurrentChannel) || {};
+    const channelIsArchived = channel.delete_at !== 0;
+
+    const toggleFavoriteCallback = useCallback((e: React.MouseEvent<HTMLButtonElement>) => {
+        e.stopPropagation();
+        if (isFavorite) {
+            dispatch(unfavoriteChannel(channel.id));
+        } else {
+            dispatch(favoriteChannel(channel.id));
+        }
+    }, [isFavorite, unfavoriteChannel, favoriteChannel, channel.id]);
```

Comment:
`favoriteChannel` and `unfavoriteChannel` are imported functions, so they don't need to be in the dependencies (they are not expected to change during the execution of the program).